vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,862 @@
+
+Common options for `ANL01FL` would be to set null for invalid assessments or
+those occurring after new anti-cancer therapy, or to only flag assessments on or
+after date of first treatment/randomization, or rules to cover the case when a
+patient has multiple observations per visit/date (e.g. by selecting the worst value).
+Another consideration could be extra potential protocol-specific sources of
+Progressive Disease such as radiological assessments, which could be
+pre-processed here to create a PD record to feed downstream derivations.
+
+For the derivation of the parameters it is expected that the subject
+identifier variables (usually `STUDYID` and `USUBJID`) and `ADT` are a
+unique key. 
+
+In the below example we consider only valid assessments and
+those occurring on or after randomization date. If there is more than
+one assessment at a date, the worst one is flagged.
+
+```{r}
+adrs_imwg <- adrs_imwg %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(!!!get_admiral_option("subject_keys"), ADT),
+      order = exprs(AVAL, RSSEQ),
+      new_var = ANL01FL,
+      mode = "first"
+    ),
+    filter = !is.na(AVAL) & AVALC != "MISSING" & ADT >= RANDDT
+  )
+```
+
+### Exclude Assessments After New Anti-Cancer Therapy (`ANL02FL`) {#anl02fl}
+
+Here is an alternative example where those records occurring after new
+anti-cancer therapy are additionally excluded (where `NACTDT` would be
+pre-derived as first date of new anti-cancer therapy.   
+In our example `NACTDT` is present in `SUPPRS` domain. If not available,
+see `{admiralonco}`
+[Creating and Using New Anti-Cancer Start Date](nactdt.html) for deriving this
+variable).
+
+```{r, eval=TRUE}
+adrs_imwg <- adrs_imwg %>%
+  mutate(
+    ANL02FL = case_when(
+      !is.na(AVAL) & ADT >= RANDDT & ADT < NACTDT ~ "Y",
+      is.na(NACTDT) ~ "Y",
+      TRUE ~ NA_character_
+    )
+  )
+```
+
+### Flag Assessments up to First PD (`ANL03FL`)  {#anl03fl}
+
+To restrict response data up to and including first reported progressive disease
+`ANL03FL` flag could be created by using `{admiral}` function
+`admiral::derive_var_relative_flag()`.
+
+```{r}
+adrs_imwg <- adrs_imwg %>%
+  derive_var_relative_flag(
+    by_vars = get_admiral_option("subject_keys"),
+    order = exprs(ADT, RSSEQ),
+    new_var = ANL03FL,
+    condition = AVALC == "PD",
+    mode = "first",
+    selection = "before",
+    inclusive = TRUE
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs_imwg,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, AVALC, ADT, ANL01FL, ANL02FL, ANL03FL)
+)
+```
+
+### Select Source Assessments for Parameter derivations
+
+For next parameter derivations we consider only Confirmed Responses (PARAMCD = "COVR"). 
+We take post-baseline records (ANL01FL = "Y") up to and including first PD (ANL02FL = "Y") and before start of new anti-cancer therapy (ANL03FL = "Y").
```

Comment:
I think anl02fl is for anti cancer therpay and anl03fl for first PD correct? I think its little confusing

Reply:
good catch, thanks!